Feature/splash/make splash

swjw13 opened the pull request:
## Related Issue

- [Issue #4 ](https://github.com/boostcampwm-2022/android11-almost-there/issues/4)
- [Issue #5 ](https://github.com/boostcampwm-2022/android11-almost-there/issues/5)
- [Issue #8 ](https://github.com/boostcampwm-2022/android11-almost-there/issues/8)

## 작업 내용 요약

- Splash 화면 구현
- Splash 화면에서 터치 문구 구현
- 메인 화면 제작
- 화면 이동시 애니메이션 구현

## Checklist

- [x] Merge 되는 브랜치가 올바른가?
- [x] Reviewers, Assignees, Labels, Projects, Milestone 설정을 했는가?
- [x] 스스로 코드 리뷰를 충분히 진행했는가?
- [x] 프로젝트의 스타일 가이드라인(컨벤션)을 준수하는가?

swjw13 committed:
:sparkles: feat: Splash 화면 추가

- Splash 화면 추가
- Button Animation 추가

Issue #4

swjw13 committed:
:sparkles: feat: 화면 이동 추가

- 메인 화면 추가, 이동 구현
- 화면 이동시에 애니메이션 구현

Issue #5, #8

swjw13 committed:
:recycle: refactor: HardCoding 정리

- Hard String들을 String Resource로 정리
- Splash 이미지의 Text Anim을 Util로 옮김

Issue #5, #8

swjw13 committed:
:recycle: refactor: 공통된 코드들을 묶음

- main 화면 공통된 요소들을 묶어줌

Issue #5, #8

tnwlsgk1002 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -15,8 +15,16 @@
+        <activity
+            android:name=".ui.MainActivity"
+            android:exported="false">
+            <meta-data
```
자동으로 생성되는 코드 같아서 없애도 될 것 같아요!

fbghgus123 commented on app/src/main/kotlin/com/woory/almostthere/ui/FindPromiseFragment.kt:
```diff
@@ -0,0 +1,59 @@
+import com.woory.almostthere.util.animLeftToRightNavOptions
+import com.woory.almostthere.util.animRightToLeftNavOption
+
+class FindPromiseFragment : Fragment() {
```
저희 BaseFragment 쓰는게 어떤가요 ?
프래그먼트 많아지면 중복되는 코드 계속 써야할 것 같습니다
가령 onDestroy ...

fbghgus123 commented on app/src/main/kotlin/com/woory/almostthere/ui/MainActivity.kt:
```diff
@@ -0,0 +1,28 @@
+        DataBindingUtil.setContentView(this, R.layout.activity_main)
+    }
+
+    private val navController: NavController by lazy {
```
by lazy를 쓰면 어떤 이점이 있나요 ?? (저는 잘 몰라서 ...)

swjw13 replied:
by lazy를 사용하면
- 해당 변수 사용이 사용되기 전까지 객체 생성을 미룰 수 있자
- lateinit var과 비교하여 val을 사용할 수 있다
- lateinit var에서 생성이 안되는 실수를 줄일 수 있다
는 장점이 있다고 생각합니다!

soulplay95 commented on app/src/main/AndroidManifest.xml:
```diff
@@ -15,8 +15,16 @@
+            android:exported="false">
+            <meta-data
+                android:name="android.app.lib_name"
+                android:value="" />
```
현재는 `meta-data` 를 사용할 일이 없어보여서 이 부분은 제거해도 괜찮아보입니다!

fbghgus123 commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,18 +1,24 @@
+        <androidx.fragment.app.FragmentContainerView
+            android:id="@+id/container_main"
+            android:name="androidx.navigation.fragment.NavHostFragment"
+            android:layout_width="match_parent"
```
match_parent는 0dp를 사용하는게 좋아보입니다

soulplay95 commented on app/src/main/res/values/themes.xml:
```diff
@@ -13,4 +13,44 @@
     </style>
+
+    <style name="Theme.AlmostThere.Splash" parent="Theme.MaterialComponents.DayNight.NoActionBar">
+        <item name="android:background">#E9F4EC</item>
```
color 리소스로 빼는 것이 좋아보여요!

soulplay95 reviewed (approved):
👍

fbghgus123 commented on app/src/main/res/layout/fragment_find_promise.xml:
```diff
@@ -0,0 +1,62 @@
+        android:layout_height="match_parent"
+        tools:context=".ui.FindPromiseFragment">
+
+        <androidx.constraintlayout.widget.ConstraintLayout
```
[merge](https://developer.android.com/develop/ui/views/layout/improving-layouts/reusing-layouts#Merge)를 사용하여 depth를 낮추는게 좋아보입니다.

그냥 색칠한 layout만 잡을거라면 View타입으로 해도 될 것 같습니다

fbghgus123 reviewed (commented):
고생하셨습니다 !

swjw13 committed:
:recycle: refactor: 코드리뷰 반영

- 중첩된 RecyclerView를 View로 변경
- Hard Coding Color를 Resource로 변경
- match_parent -> 0dp
- 필요없는 meta-data 삭제
- 이미지를 일단 png로 삽입

PR #16
